src/libs/antares/study/study.cpp

```diff
@@ -1102,100 +1102,27 @@ bool Study::thermalClusterRename(Cluster* cluster, ClusterName newName, bool)
     Area& area = *cluster->parentArea;
     if (not cluster->parentArea)
     {
-        logs.error() << "renaming thermal cluster: no parent area";
+        logs.error() << "renaming cluster: no parent area";
         return false;
     }
 
     // Checking if the area exists
-    {
-        Cluster* found;
-        if ((found = area.thermal.list.find(newID)))
-        {
-            if (found->name() != newName)
-            {
-                area.invalidateJIT = true;
-                found->setName(newName);
-                return true;
-            }
-            return false;
-        }
-    }
-
-    // We must have the scenario rules to rename properly the cluster
-    scenarioRulesLoadIfNotAvailable();
-
-    // We will temporary override the id of the area in order to have to
-    // the new name in the archive
-    // Otherwise the values associated to the area will be lost.
-    logs.info() << "  renaming thermal cluster '" << cluster->name() << "' into '" << newName
-                << "'";
-    //	ClusterName oldId   = cluster->id();
-    //	ClusterName oldName = cluster->name();
-    //	cluster->name(newName);
-    area.invalidateJIT = true;
+    Cluster* found = nullptr;
 
-    bool ret = true;
+    if (dynamic_cast<ThermalCluster*>(cluster))
```

Comment:
Using [RTTI](https://en.wikipedia.org/wiki/Run-time_type_information) is OK. But I agree with you : this function belongs in the cluster_list class.

Also, it would be possible replace the use of dynamic_cast with virtual functions.